public/app/plugins/datasource/grafana-azure-monitor-datasource/components/LogsQueryEditor/ResourceField.tsx

```diff
@@ -0,0 +1,107 @@
+import { Button, Icon, Modal } from '@grafana/ui';
+import React, { useCallback, useEffect, useState } from 'react';
+import { AzureQueryEditorFieldProps, AzureResourceSummaryItem } from '../../types';
+import { Field } from '../Field';
+import ResourcePicker from '../ResourcePicker';
+import { parseResourceURI } from '../ResourcePicker/utils';
+import { Space } from '../Space';
+
+function parseResourceDetails(resourceURI: string) {
+  const parsed = parseResourceURI(resourceURI);
+
+  if (!parsed) {
+    return undefined;
+  }
+
+  return {
+    id: resourceURI,
+    subscriptionName: parsed.subscriptionID,
+    resourceGroupName: parsed.resourceGroup,
+    name: parsed.resource,
+  };
+}
+
+const ResourceField: React.FC<AzureQueryEditorFieldProps> = ({ query, datasource, onQueryChange }) => {
+  const { resource } = query.azureLogAnalytics;
+
+  const [resourceComponents, setResourceComponents] = useState(parseResourceDetails(resource ?? ''));
+  const [pickerIsOpen, setPickerIsOpen] = useState(false);
+
+  useEffect(() => {
+    if (resource) {
+      datasource.resourcePickerData.getResource(resource).then(setResourceComponents);
```

Comment:
Lets have a chat about this later? I don't think it's too bad at the moment (Logs makes just one API call on startup, compared to Metric's 50 or whatever), and I agree it could be optimised, but I'm just not too sure about whether putting "persisted UI state" into the query object. It does feel a bit weird to me tbh.